### ISSUE TITLE
Upgrade to Gradle 3.0

### ISSUE THREAD

**Issue by geekbeast** [kryptnostic/gradles#3]
Looks like some of our dependencies will block this for a while (specifically processors and the file collection stuff).

**Comment by makosblade** [comment]
Agreed. I tried to do this a few weeks ago but it looked like more than just a quick update.
